# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ProfilerHelpers.h
# Commit: a114b9b8e6ea
# Full Hash: a114b9b8e6eaa23a445a31efc436f4e367f7874f
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-01-10 21:49:10
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Convert IDBCursor and BackgroundCursorChild to templates to increase type safety and reduce state. r=dom-workers-and-storage-reviewers,ytausky
#   
#   This also simplifies delegating calls that are dependent on the cursor type.
#   
#   Also reduce dependency on IDBCursor.h by moving enums and type traits to IDBCursorType.h
# ==============================================================================

diff -r cc3cfee4a014 -r a114b9b8e6ea dom/indexedDB/ProfilerHelpers.h
--- a/dom/indexedDB/ProfilerHelpers.h	Fri Jan 10 11:07:59 2020 +0000
+++ b/dom/indexedDB/ProfilerHelpers.h	Fri Jan 10 14:21:18 2020 +0000
@@ -12,7 +12,7 @@
 
 #include "BackgroundChildImpl.h"
 #include "GeckoProfiler.h"
-#include "IDBCursor.h"
+#include "IDBCursorType.h"
 #include "IDBDatabase.h"
 #include "IDBIndex.h"
 #include "IDBKeyRange.h"
@@ -189,18 +189,18 @@
     }
   }
 
-  explicit LoggingString(const IDBCursor::Direction aDirection) {
+  explicit LoggingString(const IDBCursorDirection aDirection) {
     switch (aDirection) {
-      case IDBCursor::Direction::Next:
+      case IDBCursorDirection::Next:
         AssignLiteral("\"next\"");
         break;
-      case IDBCursor::Direction::NextUnique:
+      case IDBCursorDirection::Nextunique:
         AssignLiteral("\"nextunique\"");
         break;
-      case IDBCursor::Direction::Prev:
+      case IDBCursorDirection::Prev:
         AssignLiteral("\"prev\"");
         break;
-      case IDBCursor::Direction::PrevUnique:
+      case IDBCursorDirection::Prevunique:
         AssignLiteral("\"prevunique\"");
         break;
       default: